### PR TITLE
[CIS-861] Attachments migration guide

### PR DESCRIPTION
**Pull request by evsaev**
**This PR** adds a `v3.2 -> v4.x` migration guide for attachments

### CODE REVIEW

File: stream-chat-swift-docs/docs/guides/attachments-migration.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+- custom payload fields are are directly on attachment thanks to `dynamicMemberLookup`
+- attachment `id` is not-optional 🎉
+
+```swift
+
+// It's recommended but not required to create a typealias for custom attachment type
+typealias ProductAttachment = _ChatMessageAttachment<Product>
+
+// Get attachments of custom type
+let productAttachments: [ProductAttachment] = message.attachments()
```
Comment (VojtaStavik) [inline stream-chat-swift-docs/docs/guides/attachments-migration.md:397]: IMO this is too magical notation :) Does the following work too?
```swift
let productAttachments = message.attachments(payloadType: ProductAttachment.self)
```

**Review by VojtaStavik — APPROVED** [review]
Great 🎉

File: stream-chat-swift-docs/docs/guides/attachments-migration.md (hunk 1 above)
Comment (evsaev) [reply stream-chat-swift-docs/docs/guides/attachments-migration.md]: Yep, it does 😄 ✅ Updated this line

**Review by b-onc — APPROVED** [review]
You're a wizard, Pavel 🦉

**Comment by VojtaStavik** [comment]
Let's merge this :shipit:

**Pull request merged**